jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/ProcessDefinitionDataSetProvider.java

```diff
@@ -0,0 +1,187 @@
+        processDefinitionDataSetDef.setServerTemplateId(consoleDataSetLookup.getServerTemplateId());
+        if (consoleDataSetLookup.getServerTemplateId() == null || consoleDataSetLookup.getServerTemplateId().isEmpty()) {
+            return buildDataSet(def,
+                                new ArrayList<>(),
+                                new ArrayList<>());
+        }
+        String defaultColumnSort = COL_ID_PROCESSNAME;
+        boolean defaultSortOrder = false;
+
+        DataSetSort dataSetSort = lookup.getFirstSortOp();
+        if (dataSetSort != null && dataSetSort.getColumnSortList().size() > 0) {
+            defaultColumnSort = dataSetSort.getColumnSortList().get(0).toString();
+            defaultSortOrder = dataSetSort.getColumnSortList().get(0).getOrder().equals(SortOrder.ASCENDING);
+        }
+
+        final StringBuffer searchText = new StringBuffer();
+        for (DataSetFilter filter : lookup.getOperationList(DataSetFilter.class)) {
+            filter.getColumnFilterList().forEach(columnFilter -> {
+                if (columnFilter instanceof CoreFunctionFilter) {
+                    CoreFunctionFilter coreFunctionFilter = (CoreFunctionFilter) columnFilter;
+                    if (coreFunctionFilter.getParameters().size() > 0) {
+                        searchText.append(coreFunctionFilter.getParameters().get(0));
+                    }
+                }
+            });
+        }
+
+        List<ProcessSummary> processSummaryList = processRuntimeDataService.getProcessesByFilter(consoleDataSetLookup.getServerTemplateId(),
+                                                                                                 searchText.toString(),
+                                                                                                 lookup.getRowOffset() / lookup.getNumberOfRows(),
+                                                                                                 lookup.getNumberOfRows(),
+                                                                                                 defaultColumnSort,
+                                                                                                 defaultSortOrder);
+
+        processSummaryList.forEach(
+                processSummary -> {
+                    dsBuilder.row(processSummary.getProcessDefName(),
+                                  processSummary.getVersion(),
+                                  processSummary.getDeploymentId(),
+                                  processSummary.getProcessDefId(),
+                                  processSummary.isDynamic());
+                }
+        );
+
+        DataSet dataSet = dsBuilder.buildDataSet();
+        dataSet.setUUID(PROCESS_DEFINITION_DATASET);
+        return dataSet;
+    }
+
+    @Override
+    public boolean isDataSetOutdated(DataSetDef def) {
+        return false;
+    }
+
+    protected DataSet buildDataSet(DataSetDef def,
```

Comment:
please review this method, most of it is not needed as it is only used once with an empty list for both parameters.